src/GenericContainer/tests/GenericContainerTestPlusYarp.cpp

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file GenericContainerTest.cpp
+ * @authors Stefano Dafarra
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+// Catch2
+#include <catch2/catch.hpp>
+#include <iDynTree/Core/VectorDynSize.h>
+#include <iDynTree/Core/VectorFixSize.h>
+#include <iDynTree/Core/TestUtils.h>
+
+#include <yarp/sig/Vector.h>
+
+#include <memory>
+
+#include <BipedalLocomotionControllers/GenericContainer.h>
+
+using namespace BipedalLocomotionControllers;
+
+TEST_CASE("Generic Container")
+{
+    SECTION("Copy")
+    {
+        iDynTree::VectorDynSize vector(5);
+        iDynTree::getRandomVector(vector);
+        GenericContainer container(iDynTree::make_span(vector));
+
+        yarp::sig::Vector copiedIn;
+        copiedIn.resize(5);
+        GenericContainer containerToBeCopied = makeGenericContainer(copiedIn);
+
+        containerToBeCopied = container;
+
+        for (long i = 0; i < container.size(); ++i)
```

Comment:
Why `long`?

Reply:
Apparently that's the ``index_type`` given by ``Span`` with ``double`` vectors. I discovered it with the static analyzer.